src/raft_server.c

```diff
@@ -4915,131 +4915,188 @@ raft_server_net_client_request_init_sm_apply(
 void
 raft_server_backend_setup_last_applied(struct raft_instance *ri,
                                        raft_entry_idx_t last_applied_idx,
+                                       raft_entry_idx_t last_applied_sub_idx,
+                                       raft_entry_idx_t last_applied_sub_idx_max,
                                        crc32_t last_applied_cumulative_crc)
 {
     NIOVA_ASSERT(ri && (raft_instance_is_booting(ri) ||
                         raft_instance_is_recovering(ri)));
 
     ri->ri_last_applied_idx = last_applied_idx;
     ri->ri_last_applied_synced_idx = last_applied_idx;
+    ri->ri_last_applied_sub_idx = last_applied_sub_idx;
+    ri->ri_last_applied_sub_idx_max = last_applied_sub_idx_max;
     ri->ri_last_applied_cumulative_crc = last_applied_cumulative_crc;
 
+
     DBG_RAFT_INSTANCE(LL_TRACE, ri, "");
 }
 
 static raft_server_epoll_sm_apply_t
 raft_server_last_applied_increment(struct raft_instance *ri,
                                    const struct raft_entry_header *reh)
 {
-    NIOVA_ASSERT(ri && reh &&
-                 (reh->reh_index == (ri->ri_last_applied_idx + 1)));
+    NIOVA_ASSERT(ri && reh && 
+        (reh->reh_num_entries > 0));
```

Comment:
The way I approach this formatting is like this since it fits on a single line:
```
 NIOVA_ASSERT(ri && reh && (reh->reh_num_entries > 0));
```

Otherwise a subclause can be placed onto a new line like:
```
NIOVA_ASSERT(ri && reh &&
             (reh->reh_num_entries_really_long_name_for_example > 0));
```

If that subclause is even longer then we will shift the alignment left:
```
NIOVA_ASSERT(
    ri && reh &&
    (reh->reh_num_entries_really_long_name_for_example_super_long > 0));
```
```